### DISCUSSION TITLE
respect the time zone from the server

### DISCUSSION THREAD

**Discussion by starklord** [serverpod/serverpod#1180] [Q&A] [answered]
**Describe the bug**
It seems the default timezone from a serverpod app is always converted to: DateTime.now().toUtc()
the problem is that every server configuration has its own timezone(Example America/Lima that is UTC-5)
so when there are logs or table columns that are by default catching the 'now()' datetime, you dont get the correct server time and it differs from what you expect to get when comparing or showing dates for example.

**To Reproduce**
1. create a postgres table column with the type: Timestamp without timezone
2. set the table column that catchs as default 'now()' (the actual datetime from the server)
3. save a new record
4. check the datetime created and it will differ from the actual datetime that the server has

**Expected behavior**
it should expect to have the datetime created with the default timezone the server has.

**Screenshots**

**Serverpod versions**
1.1.1

**Platform information**
ubuntu 22.04

**Additional context**
serverpod running directly without docker container

**Comment by vlidholt** [comment, marked as answer]
This works as expected. The idea is that you should always store times in UTC in your database, and Serverpod always write to UTC times. Mixing timezones can otherwise quickly become a mess.

You basically have two options if you want to do manual queries.

1) Set your timezone to UTC.
```sql
SET TIMEZONE='UTC';
```

2) Write your time in UTC.
```sql
INSERT INTO your_table (your_timestamp_column)
VALUES (now() AT TIME ZONE 'UTC');
```

**Comment by starklord** [comment]
Ok i understand this behaviour is because of good practices so i will keep it in mind and thank you for that explanation. Maybe i will have some troubles with legacy tables and apps but ill try to solve that scenarios one way or another.

**Comment by dkbast** [comment]
I see a problem here since "round-tripping" - that is setting a (local) DateTime on a model in the client persisting it in the server and sending it back to the client results in a time shift - what is the recommended way to handle this in the client? The client usually needs to have the DateTime in local time, not in UTC. @vlidholt

**Reply by gazialankus** [reply]
Actually there is no time shift, it's the same point in time but expressed in a different timezone (UTC). To get rid of this nuisance you would need to call .toLocal() on every DateTime that you read from the database.

**Reply by gazialankus** [reply]
I agree that It would be nice if generated code did .toLocal() for us, but I can also see other people not wanting that. We will just have to deal with it, it seems.
